exec/java-exec/src/main/java/org/apache/drill/exec/vector/complex/fn/VectorOutput.java

```diff
@@ -242,8 +247,9 @@ public void writeTimestamp(boolean isNull) throws IOException {
           ts.writeTimeStamp(dt.getMillis());
```

Comment:
I run regression tests for your branch and found one test failure:
https://github.com/mapr/drill-test-framework/blob/master/framework/resources/Datasources/complex/json/extended.json#L1

The stack trace is the following:
`java.time.format.DateTimeParseException) Text '2015-03-12T21:54:31.809+0530' could not be parsed at index 23
    java.time.format.DateTimeFormatter.parseResolved0():1949
    java.time.format.DateTimeFormatter.parse():1851
    java.time.OffsetDateTime.parse():402
    org.apache.drill.exec.vector.complex.fn.VectorOutput$MapVectorOutput.writeTimestamp():356
    org.apache.drill.exec.vector.complex.fn.VectorOutput.innerRun():112
    org.apache.drill.exec.vector.complex.fn.VectorOutput$MapVectorOutput.run():301
    org.apache.drill.exec.vector.complex.fn.JsonReader.writeMapDataIfTyped():505
    org.apache.drill.exec.vector.complex.fn.JsonReader.writeData():385
    org.apache.drill.exec.vector.complex.fn.JsonReader.writeDataSwitch():316
    org.apache.drill.exec.vector.complex.fn.JsonReader.writeToVector():257
    org.apache.drill.exec.vector.complex.fn.JsonReader.write():212
    org.apache.drill.exec.store.easy.json.JSONRecordReader.next():214
    org.apache.drill.exec.physical.impl.ScanBatch.next():172......`

I suppose you should change `DateTime` everywhere in this class (at least in writeTimestamp() method) to solve this issue.

Reply:
Update the DateUtility.isoFormatTimeStamp formatter pattern.  Use "XX" pattern to accept time zone offset specified as "Z" or "+5030"